std/range/package.d

```diff
@@ -2547,6 +2547,24 @@ pure @safe nothrow @nogc unittest
     assert(s.front == 42);
 }
 
+pure @safe unittest
+{
+    auto myIota = new class
+    {
+        int front = 0;
+        @safe void popFront(){front++;}
+        enum empty = false;
+    };
+    auto iotaPart = myIota.takeOne;
```

Comment:
Won't do, because the bug affects only InputRanges which are not a ForwardRanges.